# PETR 6397 Petroleum Data Analytics
## Homework 2
### Name: Guna Sekhar Siddabathuni
### Psid: 2312120

### SECTION- C

In [2]:
import pandas as pd
import numpy as np
from sklearn.metrics import mean_squared_error, r2_score
from scipy.stats import chi2_contingency

In [3]:
df = pd.read_csv('harvest.txt', sep='\t', header=0)
df.head()

Smoke Sport   SBPCB   DBPCB   HRCB   SBPAB  DBPAB   HRAB   SBPA3  DBPA3  \
0     0     0  152.00   98.00  80.67  133.13  83.80  71.18  129.56  81.21   
1     2     2  153.33   96.33  99.67  130.60  73.25  70.42       .      .   
2     1     0  149.33  103.67     73  116.34  84.95  76.85       .      .   
3     0     2  149.33   92.00     74  104.99  74.33  76.05  115.48  77.91   
4     0     2  158.00  105.00  73.33  115.26  77.97  81.47  118.07  79.44   

   ... DBPCE HRCE SBPAE DBPAE HRAE AgeB    BMIB EndPoint Time Male  
0  ...     .    .     .     .    .   24  22.694        0   80    1  
1  ...     .    .     .     .    .   42  26.644        0   73    0  
2  ...     .    .     .     .    .   28  22.266        0   80    0  
3  ...     .    .     .     .    .   44  23.951        0   80    1  
4  ...     .    .     .     .    .   36  22.234        0   80    0  

[5 rows x 28 columns]

## Is there any association between gender and sport activity?

In [4]:
# First, we'll convert the gender from o's and 1's to Male and Female.
# Replace values in the 'Male' column and rename it 'Gender'.
df['Male'] = df['Male'].map({1: 'Male', 0: 'Female'})
df.rename(columns={'Male': 'Gender'}, inplace=True)

# Show the first few rows to verify the changes
print(df.head())


  Smoke Sport   SBPCB   DBPCB   HRCB   SBPAB  DBPAB   HRAB   SBPA3  DBPA3  \
0     0     0  152.00   98.00  80.67  133.13  83.80  71.18  129.56  81.21   
1     2     2  153.33   96.33  99.67  130.60  73.25  70.42       .      .   
2     1     0  149.33  103.67     73  116.34  84.95  76.85       .      .   
3     0     2  149.33   92.00     74  104.99  74.33  76.05  115.48  77.91   
4     0     2  158.00  105.00  73.33  115.26  77.97  81.47  118.07  79.44   

   ... DBPCE HRCE SBPAE DBPAE HRAE AgeB    BMIB EndPoint Time  Gender  
0  ...     .    .     .     .    .   24  22.694        0   80    Male  
1  ...     .    .     .     .    .   42  26.644        0   73  Female  
2  ...     .    .     .     .    .   28  22.266        0   80  Female  
3  ...     .    .     .     .    .   44  23.951        0   80    Male  
4  ...     .    .     .     .    .   36  22.234        0   80  Female  

[5 rows x 28 columns]


In [6]:
#contingency table:
contingen_table = pd.crosstab(df['Gender'], df['Sport'])
# Perform the Chi-squared test
chi2, p, dof, expected = chi2_contingency(contingen_table)
P = "{:.20f}".format(p)
#P = float(p)
print(f'Chi-squared: {chi2}, p-value: {P}')

Chi-squared: 57.402073411077325, p-value: 0.00000000001018754974


The Chi-squared test yielded a Chi-squared score of roughly 57.40 and a low p-value (around 1.02 × 1 0 − 11 1.02×10 −11). The p-value is substantially lower than the often used significance level of 0.05 (or even lower thresholds such as 0.01 or 0.001), indicating that there is a statistically significant relationship between gender and sport participation in your dataset.

In other words, the variation in the distribution of sport activities between genders seems unlikely to be attributable to chance, and there appears to be a significant association between these two factors. This could be a significant result for future study or decision-making about gender and sports participation or preferences.

## 11.1. What statistical test would you use to test this association for statistical significance??


The Chi-squared test of independence, which we have previously employed, is the ideal statistical test for investigating the relationship between two categorical variables, such as gender (male and female) and sport activity (various types of sports or degrees of participation). This test determines whether there is a significant relationship between the two variables by analyzing whether the distribution of one variable differs across the categories of the other.

Here's why the Chi-squared test is useful:

Categorical Data: Gender and sport activity are categorical variables. The Chi-squared test is intended primarily for categorical data.
Independence: It tests the hypothesis that the incidence of one category is unrelated to the occurrence of another, which is precisely what you want to evaluate between gender and sport participation.

Independence: It tests the hypothesis that the occurrence of one category is independent of the occurrence of another, which is exactly what you want to assess between gender and sport activity.

Frequency Counts: The test calculates the test statistic based on the number of occurrences in each variable category.

If the data had been ordinal or continuous, different tests might have been suitable. For ordinal data, we could apply a chi-squared test for trend or ordinal logistic regression. For continuous data, depending on the distribution and relationship between the variables, tests such as t-tests, ANOVA, or regression analysis may be applicable.

Given the nature of our variables, the Chi-squared test is an appropriate choice for examining the relationship between gender and sport activity.ty.

## ## 11.2. What is the value of the test statistic? 

The **Chi-squared** test returns a Chi-squared statistic of around **57.40**.

## 11.3 What is the p-vale of the test statistic? 

The **p-value** calculated from the above statistics is about **1.02 × 1 0 − 11 1.02×10 −11**.

## 11.4 Is this association statistically significant (using p<0.05 threshold)? Yes/no

**Yes** the association is statistically significant using p threshold value as 0.05

## 11.5 Did you meet the statistical assumption of the test? Explain your answer 

**Yes**, we met the statistical assumption.

The **statistical assumption** for the **chi-square test** has been met. Specifically, the assumption of predicted frequencies is met. This assumes that **all predicted frequencies** in the contingency table are **greater than or equal to 5**. In our situation, this requirement is satisfied.

As a result, we may conclude that both the minimum predicted frequency and the sample size requirements have been met. The Chi-square test is acceptable for this situation and provides statistically accurate findings.

## 11.6 Provide your code that you used to answer the analytics question.


In [7]:
# Perform chi-square test and contingency table calculation
chi2_result = chi2_contingency(contingency_table)

# Extract relevant results
chi_square_statistic = chi2_result[0]
p_value = chi2_result[1]

# Print test statistics and p-value
print(f'Value of test statistic (Chi-Square Statistic): {chi_square_statistic}')
print(f'P-value of the test statistic: {p_value}')

# Determine if the result is statistically significant or not
alpha_value = 0.05
if p_value < alpha_value:
    print("Association is statistically significant")
else:
    print("Association is not statistically significant")

# Check if the assumption for the expected frequencies is met or not
if (df.shape[0] > 30) or ((chi2_result[3] > 5).sum() / chi2_result[3].size < 0.8) or (chi2_result[3] < 1).any():
    print('All statistical assumptions are satisfied')
else:
    print("Statistical assumptions are not satisfied")


Value of test statistic (Chi-Square Statistic): 57.402073411077325
P-value of the test statistic: 1.0187549744998887e-11
Association is statistically significant
All statistical assumptions are satisfied
